crates/notion-core/src/tool/mod.rs

```diff
@@ -131,6 +133,62 @@ impl ToolSpec {
             package => ToolSpec::Package(package.into(), version),
         })
     }
+
+    /// Get a valid, sorted `Vec<ToolSpec>` given a `Vec<String>`.
+    ///
+    /// Accounts for the following error conditions:
+    ///
+    /// - `notion install node 12`, where the user intended to install `node@12`
+    ///   but used syntax like in nodenv or nvm
+    /// - invalid version specs
+    ///
+    /// Returns a listed sorted so that if `node` is included in the list, it is
+    /// always first.
+    pub fn from_strings<T>(tool_strs: &mut [T], action: &str) -> Fallible<Vec<ToolSpec>>
```

Comment:
It looks like we don't need the mutability for the tool list. I checked locally and opened a PR against your branch to pull it out.

Reply:
Excellent catch!